### PR TITLE
[RW-5462][risk=no] Endpoint for COPE survey versions

### PR DESCRIPTION
**Pull request by freemabd**
This PR is adding 2 new endpoints for the COPE survey versions. There are 2 ways to call for survey versions. With question concept id or question concept id and answer concept id.

1. `findSurveyVersionByQuestionConceptId(cdrVersionId, surveyConceptId, questionConceptId)`
2. `findSurveyVersionByQuestionConceptIdAndAnswerConceptId(cdrVersionId, surveyConceptId, questionConceptId, answerConceptId)`

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/cdr/dao/CBCriteriaDaoTest.java [hunk 1]
```diff
@@ -365,4 +366,84 @@ public void findMenuOptions() {
...
+    assertThat(dbSurveyVersions.get(1).getItemCount()).isEqualTo(230);
+    assertThat(dbSurveyVersions.get(2).getSurveyId()).isEqualTo(102);
+    assertThat(dbSurveyVersions.get(2).getVersion()).isEqualTo("July 2020");
+    assertThat(dbSurveyVersions.get(2).getItemCount()).isEqualTo(181);
+    jdbcTemplate.execute("drop table cb_survey_version");
+    jdbcTemplate.execute("drop table cb_survey_attribute");
+  }
+
+  @Test
+  public void findSurveyVersionByQuestionConceptIdAndAnswerConceptId() {
```
Comment (NehaBroad) [inline api/src/test/java/org/pmiops/workbench/cdr/dao/CBCriteriaDaoTest.java:411]: Should we also add a negative test say if itemCount is null?

File: api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java [hunk 2]
```diff
@@ -249,4 +250,35 @@
           "select distinct domain_id as domain, type, is_standard as standard from cb_criteria order by domain, type, is_standard",
       nativeQuery = true)
   List<DbMenuOption> findMenuOptions();
+
+  @Query(
+      value =
+          "select surveyId, version, itemCount from( "
+              + "select csv.survey_id as surveyId, csv.version as version, sum(csa.item_count) as itemCount, csv.display_order "
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java:257]: Question: here we have sum(csa.item_count) vs just returning csa.item_count in method **findSurveyVersionByQuestionConceptIdAndAnswerConceptId** , 

Is it because  for findSurveyVersionByQuestionConceptId,  its aggregating the item_count of all answers for question_id passed?

File: api/src/test/java/org/pmiops/workbench/cdr/dao/CBCriteriaDaoTest.java (hunk 1 above)
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/cdr/dao/CBCriteriaDaoTest.java:411]: nvm i am assuming itemCount will have nullable=false constraint

**Review by NehaBroad — APPROVED** [review]

File: api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java (hunk 2 above)
Comment (freemabd) [reply api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java:257]: correct.
